node/middlewares/robots.ts

```diff
@@ -1,12 +1,49 @@
+import { Apps, Logger } from "@vtex/api"
+
 const TEN_MINUTES_S = 10 * 60
 
-export async function robots (ctx: Context) {
-  const {vtex: {account, production, platform}} = ctx
-  const { clients } = ctx
+interface RobotsFile {
+  robots: Record<string, string>
+}
+
+const SITEMAP_BUILD_FILE = 'dist/vtex.store-sitemap/build.json'
+
+const getAppRobot = (apps: Apps, deps: Record<string, string[]>, logger: Logger) => async (appName: string) => {
+  const robots = await apps.getAppJSON(appName, SITEMAP_BUILD_FILE, true)
+  if (robots && !deps[appName].includes('vtex.store-sitemap@1.x')) {
+    logger.warn({ message: `App ${appName} exports robots config, but does not depend on vtex.store-sitemap@1.x` })
+  }
+  return robots as Partial<RobotsFile> | null
+}
+
+const getRobots = async (apps: Apps, logger: Logger) => {
+  const deps = await apps.getDependencies()
+  const depList = Object.keys(deps).filter(key => !key.startsWith('infra:'))
```

Comment:
Can I filter for apps that depend on 'vtex.store-sitemap@1.x' to do less requests to apps?
Then I would remove the logger call on line 14.